go-controller/pkg/ovn/egressgw.go

```diff
@@ -93,39 +93,16 @@ func (oc *Controller) ensureRouteInfoLocked(podName ktypes.NamespacedName) (*ext
 	return routeInfo, nil
 }
 
-// getRouteInfosForGateway returns all routeInfos locked for a specific namespace and gateway IP
-func (oc *Controller) getRouteInfosForGateway(gatewayIP, namespace string) []*externalRouteInfo {
```

Comment:
The commit message seems confusion to me. This part:
"getRouteInfosForGateway() accesses a routeInfo's podExternalRoutes
without taking the routeInfo's lock."

with your previous commit it removes locking routeInfo from this function as well as getRouteInfosForNamespace. So I'm not sure how this is part of the reasoning to remove the function.

Reply:
The original code did this:

```
 		for _, route := range routeInfo.podExternalRoutes {
 			if _, ok := route[gatewayIP]; ok {
				routeInfo.Lock()
 				routeInfos = append(routeInfos, routeInfo)
 				break
 			}
```

it accessed `routeInfo.podExternalRoutes` and *then* took the routeInfo lock. But the caller of this function already does the `route[gatewayIP]` check anyway, so it's pointless to have this function now...